### PR TITLE
UI tool bar

### PR DESCRIPTION
**Pull request by amy-hoangg**
Create types for tool bar
Create a tool bar element inside div id="lightsheet"
Use Google API for icons placeholder in tool bar https://fonts.google.com/icons
Add styles for each item in tool bar
Note: the code is based on what JS Spreadsheet has

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -39,9 +39,70 @@ const columns = [
...
+    content: "format_text_wrap",
+  },
+];
 new Lightsheet(document.getElementById("lightsheet"), {
   data,
   columns,
+  toolbar,
   onCellChange: (colIndex, rowIndex, newValue) => {
     console.log(colIndex, rowIndex, newValue);
   },
```
Comment (rafinutshaw) [inline index.js:108]: The user will also provide a boolean of toolbar visibility, and the dom(optional) he wants the toolbar to mound into.
example,

const lightsheet = new Lightsheet(document.getElementById("lightsheet"), {
  data,
  columns,
 {
showToolbar: true/false,
items: toolbar,
element: document.getElementById("toolbar-dom-id")
},
  onCellChange: (colIndex, rowIndex, newValue) => {
    console.log(colIndex, rowIndex, newValue);
  },
}
//if user wants to make the toolbar visible/imvisible, he can call this method anytime
lightsheet.showToolbar(true/false);

Comment (rafinutshaw) [reply index.js:108]: You can name your methods and property names as you see fit

**Review by rafinutshaw — CHANGES_REQUESTED** [review]

**Review by rafinutshaw — APPROVED** [review]

File: src/ui/render.ts [hunk 2]
```diff
@@ -56,6 +76,37 @@ export default class UI {
...
+        const toolbarItem = document.createElement("i");
+        toolbarItem.classList.add("lightSheet_toolbar_item");
+        toolbarItem.classList.add("material-symbols-outlined");
+        toolbarItem.textContent = toolbarContent[i];
+        this.toolBarDom.appendChild(toolbarItem);
+      }
+
+      //showToolBar
+      this.showToolBar(this.toolbarOptions.showToolbar!);
+    }
```
Comment (Fluglow) [inline src/ui/render.ts:99]: This if-block can be inverted to reduce nesting.

File: src/ui/render.ts [hunk 3]
```diff
@@ -56,6 +76,37 @@ export default class UI {
...
+  }
+
+  showToolBar(isShown: boolean) {
+    if (isShown) {
+      this.toolBarDom.style.display = "flex";
+    } else {
+      this.toolBarDom.style.display = "none";
+    }
+  }
```
Comment (Fluglow) [inline src/ui/render.ts:108]: If we skip creating `this.toolBarDom` as I suggest in an earlier comment, this method should check for `toolBarOptions`. 
The method can also be simplified to `this.toolBarDom.style.display = isShown ? "flex" : "none";`

File: tests/ui/toolBar.test.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      data: [],
+    });
+    const toolBarElement = targetElementMock.querySelector(
+      ".lightsheet_table_toolbar",
+    );
+
+    // Expect that the toolbar element exists
+    expect(toolBarElement).toBeTruthy();
+    // Expect that the toolbar element has 13 children
+    expect(toolBarElement!.children.length).toBe(13);
```
Comment (Fluglow) [inline tests/ui/toolBar.test.ts:23]: Maybe compare against `ToolbarItems.length` instead?

**Review by Fluglow — COMMENTED** [review]
Left some notes, just minor things :+1:

**Comment by rafinutshaw** [comment]
> Left some notes, just minor things 👍

Good catch. Resolved them all. @Fluglow

**Pull request merged**